syntax = "proto3";

package workloads;

// Service definition for handling video streaming
service VideoStreamer {
  rpc StreamVideo (stream VideoChunk) returns (VideoResponse);
}

// Service definition for general task management
service TaskManager {
  rpc SendTask (TaskRequest) returns (TaskResponse);
  rpc StreamTask (stream TaskChunk) returns (TaskResponse);
}

// Message for video chunk data
message VideoChunk {
  bytes data = 1;
  string processing_type = 2;  // Added processing_type field
}

// Response message for video stream
message VideoResponse {
  string message = 1;
}

// Message for general task request
message TaskRequest {
  string task_id = 1;
  string task_type = 2;
  string payload = 3;
}

// Message for general task response
message TaskResponse {
  string message = 1;
  string task_id = 2;
}

// Message for task chunk data (similar to VideoChunk but for general tasks)
message TaskChunk {
  bytes data = 1;
  string task_id = 2;
}
